cli/proteus/main.go

```diff
@@ -120,28 +121,67 @@ func genAll(c *cli.Context) error {
 	}
 
 	for _, p := range packages {
-		outPath := filepath.Join(goSrc, p)
+		outPath := goSrc
 		proto := filepath.Join(path, p, "generated.proto")
-		cmd := exec.Command(protocPath,
-			fmt.Sprintf(
-				"--proto_path=%s:%s:%s:.",
-				goSrc,
-				filepath.Join(protobufSrc, "protobuf"),
-				filepath.Join(path, p),
-			),
-			fmt.Sprintf("--gofast_out=plugins=grpc:%s", outPath),
-			proto,
-		)
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		if err := cmd.Run(); err != nil {
+
+		err := protocExec(protocPath, p, outPath, proto)
+		if err != nil {
 			return fmt.Errorf("error generating Go files from %q: %s", proto, err)
 		}
+
+		matches, err := filepath.Glob(filepath.Join(path, p, "*.pb.go"))
+		if err != nil {
+			return fmt.Errorf("error moving Go files")
+		}
+
+		moveToDir := filepath.Join(outPath, p)
+		for _, s := range matches {
+			mv(s, moveToDir)
+		}
 	}
 
 	return genRPCServer(c)
```

Comment:
this should be inside the if. If the protoc generation failed there's no point in generating the RPC.